libraries-5/src/main/java/com/baeldung/yauaa/UserAgentAttributeLoggingFilter.java

```diff
@@ -0,0 +1,45 @@
+package com.baeldung.yauaa;
+
+import java.io.IOException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import nl.basjes.parse.useragent.UserAgent;
+import nl.basjes.parse.useragent.UserAgentAnalyzer;
+
+@Component
+public class UserAgentAttributeLoggingFilter extends OncePerRequestFilter {
+
+    private static final Logger log = LoggerFactory.getLogger(UserAgentAttributeLoggingFilter.class);
+    private final UserAgentAnalyzer userAgentAnalyzer;
+
+    public UserAgentAttributeLoggingFilter(UserAgentAnalyzer userAgentAnalyzer) {
+        this.userAgentAnalyzer = userAgentAnalyzer;
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+        throws ServletException, IOException {
+        String userAgentString = request.getHeader("User-Agent");
+
+        if (StringUtils.hasText(userAgentString)) {
```

Comment:
Does this mean that .parse can't cope with empty/null strings?

Reply:
It can, I've removed this check.